service/verify.go

```diff
@@ -0,0 +1,170 @@
+package main
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/slsa-framework/slsa-verifier/cmd"
+	"github.com/slsa-framework/slsa-verifier/verification"
+)
+
+var errInvalid = errors.New("invalid")
+
+type v1Query struct {
+	// Compulsory fields.
+	Source       string `json:"source"`
```

Comment:
+1. I think one big mistake in cosign's API was that the source was never pinned. We should never try signature verification without a pinned subject.